### PR TITLE
Update dependency versions and support new @hapi package scope versions

### LINKED ISSUE #24: New version to support @hapi/good
hapi is moving to using scoped packages (https://www.npmjs.com/package/@hapi/hapi) and upgrading breaks the peer dependency check.

I'm not sure if you can conditionally target two packages of different names, so this may require creating a new major version that tracks the @hapi packages in parallel.

### PR DESCRIPTION
**Pull request by PizzaBrandon**
Resolves #24 

This brings all packages up to the latest version and moves to supporting the new @hapi npm package scope.

Additional changes were made to correct lint errors and support new async/await interfaces in `lab` and `wreck`.

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -1,2 +1,2 @@
 language: node_js
-node_js: 4.0
+node_js: 8.0
```
Comment (nakardo) [inline .travis.yml]: Looks like tests are not being run. Try changing this to `node_js: 8` so it matches a newest version, I think you may have introduced some language feature on your changes missing on `8.0`.

File: lib/index.js [hunk 2]
```diff
@@ -207,7 +207,9 @@ class GoodSlack extends Stream.Writable {
             payload: Stringify(payload)
         };
 
-        Wreck.post(this._config.url, data, callback);
+        Wreck.post(this._config.url, data)
+            .then((result) => callback(null, result))
+            .catch((error) => callback(error));
```
Comment (nakardo) [inline lib/index.js]: Could be expressed as: `.catch(callback);`

**Review by nakardo — CHANGES_REQUESTED** [review]
Overall looks good, I left a few comments. Please double check if tests passes after making the change I suggested and I'll make the merge.

File: lib/index.js (hunk 2 above)
Comment (PizzaBrandon) [reply lib/index.js]: I included it this way for specificity and to match the format of the line above, but can change.

**Comment by PizzaBrandon** [comment]
Tests actually ran that time.

**Comment by nakardo** [comment]
Thanks for working on this. I've just published your changes.

**Pull request merged**